src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java

```diff
@@ -250,11 +250,22 @@ public String getRemote() {
     }
 
     @Override
+    @CheckForNull
     protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOException, InterruptedException {
-        // TODO reimplement using GitHub API
-        return super.retrieve(head, listener);
+        StandardCredentials credentials = Connector.lookupScanCredentials(getOwner(), apiUri, scanCredentialsId);
+        if (credentials == null) {
+            listener.getLogger().println("No scan credentials, skipping");
+            return null;
+        }
+        listener.getLogger().format("Connecting to %s using %s%n", getDescriptor().getDisplayName(), CredentialsNameProvider.name(credentials));
+        GitHub github = Connector.connect(apiUri, credentials);
```

Comment:
Sorry what does this method do? If you are using github-plugin integration, then you don't need it's credentials as they should be set only globally. 

If it about git, then github token can't work with `git://`